synapse/lib/stormtypes.py

```diff
@@ -644,53 +644,54 @@ class LibDmon(Lib):
          'type': {'type': 'function', '_funcname': '_libDmonAdd',
                   'args': (
                       {'name': 'text', 'type': ['str', 'storm:query'],
-                       'desc': 'The Storm query to execute in the Dmon loop.', },
-                      {'name': 'name', 'type': 'str', 'desc': 'The name of the Dmon.', 'default': 'noname', },
+                       'desc': 'The Storm query to execute in the Dmon loop.'},
+                      {'name': 'name', 'type': 'str', 'desc': 'The name of the Dmon.', 'default': 'noname'},
+                      {'name': 'ddef', 'type': 'dict', 'desc': 'Additional daemon definition fields. ', 'default': None},
```

Comment:
We should probably add a Dmon schema that is checked somewhere.

Reply:
schema addition should be checked in the top half of the dmon handler

Reply:
schema exists and is checked by the bottom half of the handler ( which would need to be the case anyway ). Top half checking is mostly for avoiding the nexus log entry which in this case is pretty minor, but feel free to add a ticket about it. :+1: